optuna/testing/storages.py

```diff
@@ -98,7 +98,17 @@ def __enter__(
             self.thread = threading.Thread(target=self.server.start)
             self.thread.start()
 
-            return GrpcStorageProxy(host="localhost", port=port)
+            proxy = GrpcStorageProxy(host="localhost", port=port)
+
+            # Wait until the server is ready.
+            while True:
+                try:
+                    proxy.get_all_studies()
+                    break
+                except grpc.RpcError:
+                    continue
```

Comment:
Could you add a `time.sleep()` to avoid a busy loop?

```suggestion
            while True:
                try:
                    proxy.get_all_studies()
                    break
                except grpc.RpcError:
                    time.sleep(1)
                    continue
```